content/docs/command-reference/push.md

```diff
@@ -274,3 +274,90 @@ Cache and remote 'r1' are in sync.
 
 And running `dvc status --cloud`, DVC verifies that indeed there are no more
 files to push to remote storage.
+
+## Example: [Version-aware] remote for readable storage
+
+[version-aware]:
+  /doc/user-guide/data-management/cloud-versioning#version-aware-remotes
+
+Let's set up a `version-aware` remote, which uses cloud versioning to organize
+the remote storage.
+
+```cli
+$ dvc remote add --force --default r1 s3://mybucket
+
+$ dvc remote modify r1 version_aware true
+```
+
+Push to the `version-aware` remote:
+
+```cli
+$ dvc push
+```
+
+Now let's look at what was pushed to the remote. Unlike the [example above], the
+version-aware remote looks similar to the data in your workspace and is easy to
+read.
+
+[example above]: #example-what-happens-in-the-cache
+
+```cli
+# Show the current versions.
+$ aws s3 ls --recursive s3://mybucket/
+
+2023-02-01 15:24:09    1708591 data/prepared/test.tsv
+2023-02-01 15:24:10    6728772 data/prepared/train.tsv
+
+# Show all object versions.
+$ aws s3api list-object-versions --bucket mybucket
+{
+    "Versions": [
+        {
+            "ETag": "\"b656f1a8273d0c541340cb129fd5d5a9\"",
+            "Size": 1708591,
+            "StorageClass": "STANDARD",
+            "Key": "data/prepared/test.tsv",
+            "VersionId": "T6rFr7NSHkL3v9tGStO7GTwsVaIFl42T",
+            "IsLatest": true,
+            "LastModified": "2023-02-01T20:24:09.000Z",
+            ...
+        },
+        {
+            "ETag": "\"9ca281786366acca17632c27c5c5cc75\"",
+            "Size": 6728772,
+            "StorageClass": "STANDARD",
+            "Key": "data/prepared/train.tsv",
+            "VersionId": "XaYsHQHWK219n5MoCRe.Rr7LeNbbder_",
+            "IsLatest": true,
+            "LastModified": "2023-02-01T20:24:10.000Z",
+            ...
+        }
+    ]
+```
+
+Finally, let's look at the corresponding entries in `dvc.lock`. The `version_id`
+of each file from above is recorded, so that they can be recovered even if
+overwritten by a new version.
+
+```yaml
+stages:
+  prepare:
+    ...
+    outs:
+    - path: data/prepared
+      files:
+      - size: 1708591
+        md5: b656f1a8273d0c541340cb129fd5d5a9
+        relpath: test.tsv
+        cloud:
+          r1:
+            etag: b656f1a8273d0c541340cb129fd5d5a9
+            version_id: T6rFr7NSHkL3v9tGStO7GTwsVaIFl42T
+      - size: 6728772
+        md5: 9ca281786366acca17632c27c5c5cc75
+        relpath: train.tsv
+        cloud:
+          r1:
+            etag: 9ca281786366acca17632c27c5c5cc75
+            version_id: XaYsHQHWK219n5MoCRe.Rr7LeNbbder_
```

Comment:
```suggestion
    ...
    outs:
    - path: data/prepared
      files:
      - size: 1708591
        md5: b656f1a8273d0c541340cb129fd5d5a9
        relpath: test.tsv
        cloud:
          versioned_store:
            etag: b656f1a8273d0c541340cb129fd5d5a9
            version_id: T6rFr7NSHkL3v9tGStO7GTwsVaIFl42T
      - size: 6728772
        md5: 9ca281786366acca17632c27c5c5cc75
        relpath: train.tsv
        cloud:
          versioned_store:
            etag: 9ca281786366acca17632c27c5c5cc75
            version_id: XaYsHQHWK219n5MoCRe.Rr7LeNbbder_
```

Not sure we need all this in the example though, seems too deep. Maybe just mention `version_id` and/or consider expanding the dvc.lock sample in https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#dvclock-file instead.

Reply:
p.s. yeah isn't that file missing `version_id`, actually?